core/deployment/src/main/java/io/quarkus/deployment/dev/testing/TestSupport.java

```diff
@@ -143,46 +148,36 @@ public void start() {
         }
     }
 
+    private static Pattern getCompiledPatternOrNull(Optional<String> patternStr) {
+        return patternStr.isPresent() ? Pattern.compile(patternStr.get()) : null;
```

Comment:
Same same but different:
```suggestion
        return patternStr.map(Pattern::compile).orElse(null);
```